plugins/in_winevtlog/winevtlog.c

```diff
@@ -0,0 +1,786 @@
+                                               0,
+                                               NULL) == 0)
+                                FormatMessageW(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM |
+                                               FORMAT_MESSAGE_IGNORE_INSERTS,
+                                               NULL,
+                                               status,
+                                               MAKELANGID(LANG_ENGLISH, SUBLANG_ENGLISH_US),
+                                               (WCHAR*)(&lpMsgBuf),
+                                               0,
+                                               NULL);
+                            error_message = convert_wstr((WCHAR*)lpMsgBuf, CP_ACP);
+                            flb_error("Failed to get message with %d, err = %s", status, error_message);
+                            flb_free(error_message);
+
+                            wMessage = _wcsdup((WCHAR*)lpMsgBuf);
+                            LocalFree(lpMsgBuf);
+
+                            goto cleanup;
+                        }
+                    }
+
+                    if (status != ERROR_INSUFFICIENT_BUFFER) {
+                        flb_error("failed with %d", status);
+                        goto cleanup;
+                    }
+                }
+            }
+        }
+    }
+
+    wMessage = _wcsdup(pBuffer);
+
+cleanup:
+    if (pBuffer)
+        flb_free(pBuffer);
+
+    return wMessage;
+}
+
+PWSTR get_description(EVT_HANDLE handle, LANGID langID, unsigned int *message_size)
+{
+#define BUFSIZE 256
+    WCHAR *wBuffer[BUFSIZE];
+    PEVT_VARIANT pValues = NULL;
+    DWORD dwBufferUsed = 0;
+    DWORD status = ERROR_SUCCESS, count = 0;
+    WCHAR *wMessage = NULL;
+    EVT_HANDLE hMetadata = NULL;
+
+    PCWSTR eventProperties[] = { L"Event/System/Provider/@Name" };
+    EVT_HANDLE hContext =
+            EvtCreateRenderContext(1, eventProperties, EvtRenderContextValues);
+    if (hContext == NULL) {
+        flb_error("Failed to create renderContext");
+        goto cleanup;
+    }
+
+    if (EvtRender(hContext,
+                  handle,
+                  EvtRenderEventValues,
+                  BUFSIZE,
+                  wBuffer,
+                  &dwBufferUsed,
+                  &count) != FALSE){
+        status = ERROR_SUCCESS;
+    }
+    else {
+        status = GetLastError();
+    }
+
+    if (status != ERROR_SUCCESS) {
+        flb_error("failed to query RenderContextValues");
+        goto cleanup;
+    }
+    pValues = (PEVT_VARIANT)wBuffer;
+
+    hMetadata = EvtOpenPublisherMetadata(
+            NULL, // TODO: Remote handle
+            pValues[0].StringVal,
+            NULL,
+            MAKELCID(langID, SORT_DEFAULT),
+            0);
+    if (hMetadata == NULL) {
+        goto cleanup;
+    }
+
+    wMessage = get_message(hMetadata, handle, message_size);
+
+#undef BUFSIZE
```

Comment:
Why do we want to undefine this?

Reply:
Fixed in https://github.com/fluent/fluent-bit/pull/4179/commits/a6583c0c04c813dc128074f8223dd90a3777ce78